File: Accounting/src/main/java/com/experiment/accounting/utils/ConstantUtils.java
package com.experiment.accounting.utils;

import com.experiment.accounting.entity.PaymentTypeEntity;
import com.experiment.accounting.entity.TransactionTypeEntity;
import com.experiment.accounting.exception.BusinessExceptions;

import java.util.ArrayList;
import java.util.List;

public class ConstantUtils {
    private static final List<ConstantReference> constants = new ArrayList<>();

    static {
        ConstantReference paymentTypeReference = new ConstantReference();
        ConstantReference transactionTypeReference = new ConstantReference();

        paymentTypeReference.setClazz(PaymentTypeEntity.class);
        paymentTypeReference.setName("payment-type");
        paymentTypeReference.setDesc("Payment Type");

        transactionTypeReference.setClazz(TransactionTypeEntity.class);
        transactionTypeReference.setName("transaction-type");
        transactionTypeReference.setDesc("Transaction Type");

        constants.add(paymentTypeReference);
        constants.add(transactionTypeReference);
    }

    public static ConstantReference getReferenceByName(String name){
        return constants.stream().filter(constantHierarchy -> constantHierarchy.name.equals(name)).findFirst().orElseThrow(() -> BusinessExceptions.CONSTANT_NAME_IS_NOT_ACCEPTABLE);
    }

}

File: Inventory/src/main/kotlin/com/experiment/inventory/utils/ConstantUtils.kt
package com.experiment.inventory.utils

import com.experiment.inventory.entity.PcsTypeEntity
import com.experiment.inventory.entity.ProductTypeEntity
import com.experiment.inventory.entity.base.BaseConstantEntity
import com.experiment.inventory.exception.BusinessException

class ConstantUtils {

    companion object{
        private val constants: MutableList<ConstantReference<out BaseConstantEntity>> = ArrayList()
        private var productTypeReference = ConstantReference<ProductTypeEntity>()
        private var pcsTypeReference = ConstantReference<PcsTypeEntity>()

        init{
            productTypeReference.clazz = ProductTypeEntity::class
            productTypeReference.name = "product-type"
            productTypeReference.desc = "Product Type"

            pcsTypeReference.clazz = PcsTypeEntity::class
            pcsTypeReference.name = "pcs-type"
            pcsTypeReference.desc = "Pcs Type"

            constants.add(productTypeReference)
            constants.add(pcsTypeReference)
        }

        fun getReferenceByName(name: String?): ConstantReference<out BaseConstantEntity> {
            return constants.stream().filter{ constantHierarchy: ConstantReference<out BaseConstantEntity> ->
                constantHierarchy.name.equals(
                    name
                )
            }.findFirst().orElseThrow{ BusinessException("Constant name is not acceptable")}
        }
    }
}